code/modules/ninja/energy_katana.dm

```diff
@@ -40,13 +39,12 @@
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
 
-/obj/item/energy_katana/attack_self(mob/user)
-	dash_toggled = !dash_toggled
-	to_chat(user, "<span class='notice'>You [dash_toggled ? "enable" : "disable"] the dash function on [src].</span>")
-
 /obj/item/energy_katana/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(dash_toggled && !Adjacent(target) && !target.density)
+
+	var/list/params = params2list(click_parameters)
+
+	if(params["right"] && !Adjacent(target) && !target.density)
```

Comment:
If `right click to do a thing` is going to be a new standard, perhaps we should just have a proc on the base atom/datum level that already does this fiddliness with the params?

Reply:
I'm cool with it, @Qustinnus do you have any comments on that? Making an `/afterattack_alternate` and `/attackby_alternate` etc?

Reply:
Splitting this into its own PR.